stock_orderpoint_mto_as_mts/models/product_product.py

```diff
@@ -0,0 +1,157 @@
+            "product_max_qty": 0,
+            "location_id": warehouse._get_locations_for_mto_orderpoints().id,
+            "product_uom": self.uom_id.id,
+        }
+
+    def _ensure_default_orderpoint_for_mto(self):
+        """Ensure that a default orderpoint is created for the MTO products.
+
+        Perform this on all product companies.
+        """
+        _logger.debug("Ensure orderpoint for MTO")
+        products_by_company = self._read_group(
+            domain=[("is_mto", "=", True), ("purchase_ok", "=", True)],
+            groupby=["company_id"],
+            aggregates=["id:recordset"],
+        )
+        if not products_by_company:
+            return
+        wh_obj = self.env["stock.warehouse"]
+        op_obj = self.env["stock.warehouse.orderpoint"]
+        all_mto_wh = wh_obj.search([("mto_as_mts", "=", True)])
+        op_vals_list = []
+        for company, products in products_by_company:
+            if company:
+                mto_wh = all_mto_wh.filtered(
+                    lambda w, company=company: w.company_id == company
+                )
+            else:
+                mto_wh = all_mto_wh
+            locations = self.env["stock.location"].browse()
+            for wh in mto_wh:
+                locations |= wh._get_locations_for_mto_orderpoints()
+            # Reactivate inactive orderpoints
+            inactive_ops = op_obj.with_context(active_test=False).search(
+                [
+                    ("active", "=", False),
+                    ("location_id", "in", locations.ids),
+                    ("product_id", "in", products.ids),
+                ]
+            )
+            if inactive_ops:
+                inactive_ops.write(
+                    {"active": True, "product_min_qty": 0.0, "product_max_qty": 0.0}
+                )
+            # Find products having an orderpoint for that wh
+            products_by_wh = dict(
+                op_obj._read_group(
+                    domain=[
+                        ("product_id", "in", products.ids),
+                        ("warehouse_id", "in", mto_wh.ids),
+                    ],
+                    groupby=["warehouse_id"],
+                    aggregates=["product_id:recordset"],
+                )
+            )
+            # Prepare missing orderpoints
+            for warehouse in mto_wh:
+                _logger.debug(f"Prepare orderpoint for warehouse {warehouse.name}")
+                missing_products = products - products_by_wh.get(
+                    warehouse, products.browse()
+                )
+                for product in missing_products:
+                    op_vals_list.append(
+                        product._prepare_missing_orderpoint_vals(warehouse)
+                    )
+        chunck_size = models.INSERT_BATCH_SIZE * 20
+        _logger.debug(
+            f"Create orderpoint for MTO: {len(op_vals_list)} to create - "
+            f"{len(op_vals_list)//chunck_size+1} chucks"
```

Comment:
chunk and not chunck
```suggestion
            f"{len(op_vals_list)//chunk_size+1} chunks"
```